api/src/test/java/org/openmrs/test/BaseContextSensitiveTest.java

```diff
@@ -601,12 +602,27 @@ public void setAutoIncrementOnTablesWithNativeIfNotAssignedIdentityGenerator() t
 	}
 
 	/**
+	 * Drops the not null constraint from the the specified column in the specified table
+	 *
+	 * @param columnName the column from which to remove the constraint
+	 * @param tableName the table that contains the column
+	 * @throws SQLException
+	 */
+	protected void dropNotNullConstraint(String tableName, String columnName) throws SQLException {
+		if (!useInMemoryDatabase()) {
+			throw new RuntimeException("Altering column nullability is not supported for a non in-memory database");
+		}
+		final String sql = "ALTER TABLE " + tableName + " ALTER COLUMN " + columnName + " SET NULL";
+		DatabaseUtil.executeSQL(getConnection(), sql, false);
+	}
+	/**
+
 	 * Note that with the H2 DB this operation always commits an open transaction.
```

Comment:
What's this?

Reply:
Have resolved it thanks

Reply:
@sherrif10 what is this newly added `dropNotNullConstraint(String, String)`?

The change that is expected is just to turn two `private` methods into `protected`.

Reply:
This was due to cherry-picked commit. i thought we can also inherit this commit to since this was not committed in 2.3.x as you [https://github.com/openmrs/openmrs-core/blob/2.3.x/api/src/test/java/org/openmrs/test/BaseContextSensitiveTest.java).

Reply:
Resolved thanks